core/src/main/resources/error/error-classes.json

```diff
@@ -3,6 +3,34 @@
     "message" : [ "Field name %s is ambiguous and has %s matching fields in the struct." ],
     "sqlState" : "42000"
   },
+  "CANNOT_CAST_DATATYPE" : {
+    "message" : [ "Cannot cast %s to %s." ],
+    "sqlState" : "22018"
+  },
+  "CANNOT_CHANGE_DECIMAL_PRECISION" : {
+    "message" : [ "%s cannot be represented as Decimal(%s, %s)." ],
+    "sqlState" : "22023"
+  },
+  "CANNOT_EVALUATE_EXPRESSION" : {
+    "message" : [ "Cannot evaluate expression: %s: %s" ],
```

Comment:
This is a little awkward in the case that there is no value for the first param (it would look like `Cannot evaluate expression: : expression`).

I think we should just have it be: `Cannot evaluate expression: %s` and pass `methodName: unEvaluable.toString` as the single param.